### PR TITLE
improve contract call(), send() method.

### PR DESCRIPTION
**Pull request by sirano11**
## Proposed changes

This PR improve execute smart contract method.
  - `Contract` instance can execute contract method directly.
  - It is no longer necessary to send a list object by applying a variable parameter of Object type to the smart contract function parameter.
    - contract.send("methodName", contract method arguments.....) 
    - contract.send(SendOptions, "methodName", contract method arguments.....) 
    - contract.send(SendOptions, TransactionReceiptProcessor, "methodName", contract method arguments)
    - contract.sendWithTypeReference("methodName", contract method arguments.....) 
    - contract.sendWithTypeReference(SendOptions, "methodName", contract method arguments.....) 
    - contract.sendWithTypeReference(SendOptions, TransactionReceiptProcessor,, "methodName", contract method arguments)
     - contract.call("methodName", contract method arguments....)
     - contract.call(CallObject, "methodName", contract method arguments....)
     - contract.callWithSolidityTypeReference("methodName", contract method arguments....)
     - contract.callWithSolidityTypeReference(CallObject, "methodName",  contract method arguments....)
  - Also, add newly DeployContractParams constructor  
     - constructor with variable parameters applied.
     - constructor with contract's binary data only.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-java/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-java)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: core/src/main/java/com/klaytn/caver/contract/Contract.java [hunk 1]
```diff
@@ -205,6 +207,198 @@ public KlayLogs getPastEvent(String eventName, KlayLogFilter filterOption) throw
...
+     * @return TransactionReceiptData
+     * @throws NoSuchMethodException
+     * @throws TransactionException
+     * @throws IOException
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     * @throws ClassNotFoundException
+     */
+    public TransactionReceipt.TransactionReceiptData send(String methodName, SendOptions options, Object... methodArguments) throws NoSuchMethodException, TransactionException, IOException, InstantiationException, IllegalAccessException, InvocationTargetException, ClassNotFoundException {
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/contract/Contract.java:315]: Should ʻObject...methodArguments` be passed as the last parameter unconditionally?

File: core/src/main/java/com/klaytn/caver/contract/Contract.java [hunk 2]
```diff
@@ -205,6 +207,198 @@ public KlayLogs getPastEvent(String eventName, KlayLogFilter filterOption) throw
...
+     * @return List
+     * @throws NoSuchMethodException
+     * @throws IOException
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     * @throws ClassNotFoundException
+     */
+    public List<Type> callWithSolidityWrapper(String methodName, Type... methodArguments) throws NoSuchMethodException, IOException, InstantiationException, IllegalAccessException, InvocationTargetException, ClassNotFoundException {
+        return call(methodName, CallObject.createCallObject(), methodArguments);
```
Comment (kjhman21) [inline core/src/main/java/com/klaytn/caver/contract/Contract.java:260]: @sirano11 Shouldn't we call `callWithSolidityWrapper`?

Comment (sirano11) [reply core/src/main/java/com/klaytn/caver/contract/Contract.java]: I fix it.

**Review by jimni1222 — DISMISSED** [review]
LGTM ~~ =)

File: core/src/main/java/com/klaytn/caver/contract/Contract.java [hunk 3]
```diff
@@ -205,6 +207,198 @@ public KlayLogs getPastEvent(String eventName, KlayLogFilter filterOption) throw
...
+     * @param methodArguments The arguments that need to execute smart contract method.
+     * @return List
+     * @throws NoSuchMethodException
+     * @throws IOException
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     * @throws ClassNotFoundException
+     */
+    public List<Type> callWithSolidityTypeReference(String methodName, Type... methodArguments) throws NoSuchMethodException, IOException, InstantiationException, IllegalAccessException, InvocationTargetException, ClassNotFoundException {
```
Comment (kjhman21) [inline core/src/main/java/com/klaytn/caver/contract/Contract.java:259]: @sirano11 What about just use `callWithSolidityType`? 'type' and 'reference' together makes me confused.

**Review by jimni1222 — APPROVED** [review]

**Review by jimni1222 — DISMISSED** [review]

File: core/src/main/java/com/klaytn/caver/contract/Contract.java [hunk 4]
```diff
@@ -106,6 +108,45 @@ public Contract(Caver caver, String abi, String contractAddress) throws IOExcept
         setDefaultSendOptions(new SendOptions());
     }
 
+    /**
+     * Deploy a contract
+     * @param sendOptions A SendOption instance.
+     * @param contractBinaryData A smart contract binary data.
+     * @param constructorParams A List of smart contract constructor parameter.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/contract/Contract.java:115]:
```suggestion
     * @param constructorParams The smart contract constructor parameters.
```

File: core/src/main/java/com/klaytn/caver/contract/Contract.java [hunk 5]
```diff
@@ -106,6 +108,45 @@ public Contract(Caver caver, String abi, String contractAddress) throws IOExcept
...
+        ContractDeployParams deployParams = new ContractDeployParams(contractBinaryData, Arrays.asList(constructorParams));
+        return deploy(deployParams, sendOptions);
+    }
+
+    /**
+     * Deploy a contract
+     * @param sendOptions A SendOption instance
+     * @param receiptProcessor A TransactionReceiptProcessor instance.
+     * @param contractBinaryData A smart contract binary data.
+     * @param constructorParams A List of smart contract constructor parameter.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/contract/Contract.java:135]:
```suggestion
     * @param constructorParams The smart contract constructor parameters.
```

**Review by jimni1222 — APPROVED** [review]
LGTM :)

**Review by kjhman21 — APPROVED** [review]

**Pull request merged**